### PR TITLE
OJ-2632: Adds custom metrics for response latency

### PR DESCRIPTION
**Pull request by Caitlin-cooling**
### What changed

Added custom metrics and info log for response latency in the matching lambda.

### Why did it change

This allows us to separate the response times of third parties (or stubs) from the overall response times. 

DT board to follow once the metric has been merged.

### Screenshots

image
image

### Issue tracking

- [OJ-2632](https://govukverify.atlassian.net/browse/OJ-2632)

### Environment variables or secrets

- [ ] Documented in the [README](./blob/main/README.md)
- [x] Added to deployment repository
- [ ] Added to local startup repository

[OJ-2632]: https://govukverify.atlassian.net/browse/OJ-2632?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by surajkumar — COMMENTED** [review]
(Theory) Could this not have been done purely in CloudFormation? 

Here's an example:

```
MatchingFunctionLatencyMetricFilter:
  Type: AWS::Logs::MetricFilter
  Properties:
    LogGroupName: !Ref MatchingFunctionLogGroup
    FilterPattern: "[timestamp=*Z, request_id=\"*-*\", event, REPORT, RequestId, Duration: * ms]"
    MetricTransformations:
      - MetricValue: "$.latency"
        MetricName: !Sub ${MatchingFunction}-Latency
        MetricNamespace: !Sub ${AWS::StackName}/LogMessages
```

And that's it - I haven't tested the solution but it seems like the single resource would have been better. Since in the logs we get:
```
REPORT RequestId: 0635f152-c4bc-4188-94ae-95f8455d3a2a Duration: 371.45 ms	Billed Duration: 372 ms	Memory Size: 128 MB	Max Memory Used: 128 MB
```
You could extract that out. 

Let me know what you think.

**Review by surajkumar — DISMISSED** [review]
LGTM - approved if the CF way mentioned isn't applicable/does not work

**Comment by jkunle** [comment]
Yeah if we could achieve it in cloudformation it would be good, I think we should explore this

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -1294,7 +1295,7 @@ Resources:
       FilterPattern: '{$.type = "ExecutionFailed"}'
       MetricTransformations:
         - MetricValue: "1"
-          MetricName: "NinoCheckStateMachine}-Error"
+          MetricName: "NinoCheckStateMachine-Error"
```
Comment (CharlesIC) [inline infrastructure/template.yaml:1298]: Nice catch 👌

**Review by CharlesIC — DISMISSED** [review]

**Review by surajkumar — APPROVED** [review]

**Review by CharlesIC — APPROVED** [review]

**Pull request merged**